tests/unit/s2n_record_read_test.c

```diff
@@ -164,12 +164,20 @@ int main(int argc, char *argv[])
         };
     };
 
+    const char *security_policy_test_cases[] = {
+        /* TLS 1.3 */
+        "20240503",
+        /* TLS 1.2 */
+        "20240501",
+    };
```

Comment:
You might want to assert that you really are negotiating the version you expect. Which would mean you'd probably want something like this instead:
```
struct {
    const char *policy;
    uint8_t version;
} test_cases[] = {
    { .policy = 20240503, .version = S2N_TLS13 }
}
```